### DISCUSSION TITLE
Champ date

### DISCUSSION THREAD

**Discussion by GigMatt** [MTESSDev/FRW#27] [Annonces]
Bonjour,
Question:
- Est-ce possible d'avoir une validation pour une date quelconque, ajouter 210 jours donnant le résultat d'une date X, pour ensuite inscrire dans une zone texte une date Y  (représente la différence entre X - "aujourd'hui")?

Exemple : Nous sommes le 15 mars. J'inscris la date du 28 février. (28 février + 210 jours = 26 septembre. 26 septembre - 15 mars = 195 jours) Nous aviserons le client l'attente de 195 jours.

Est-ce possible?
Je vous remercie et bonne journée

**Comment by j0ber** [comment]
Bonjour,
De ce que j'en comprend il ne s'agit pas d'une validation mais bien de l'affichage d'un nombre de jours dans une zone de texte pour aviser le client? Il n'y a pas d'erreur sur le champ date en fonction du résultat du calcul?
Merci!

**Reply by GigMatt** [reply]
Tu comprends très bien, il ne s'agit pas d'une validation (my bad), mais bien d'un calcul.

**Comment by j0ber** [comment]
J'ai complètement revu notre page qui expliquait l'injection de scripts. Vous n'aurez pas le choix pour votre besoin.

Je crois que cela répondra exactement à votre besoin et aussi que vous comprendrez l'étendue des possibilités que ça vous offre :) 


[Voir l'exemple](https://formulaires.it.mtess.gouv.qc.ca/Form/7/P700U/0/N/#p=sectionInjecterJs)

**Reply by GigMatt** [reply]
Bonjour, je trouve génial votre explication dans le P700U. On devient accro à FRW en effet :)
Seulement, le java script s'insère au debut de l'écriture du code et ce, avant le : 

accesAnonyme: true
afficherMessageSauvegardeCourrielReprise: true

Malheureusement, j'ai toujours la ligne " config: " surligné en rouge, ce qui ne me permet pas de réaliser le script. Voici un Screenshot.

![image](https://user-images.githubusercontent.com/126168415/227614223-965afd4d-c973-4ea6-965c-fb880cc74b01.png)

Nous avons "Method", nous avons "computed" devons nous aussi avoir un "watch"

Merci, je tente de rendre à succès cette fonction génial.
Je vous remercie

**Reply by anisite** [reply]
Est-ce que je pourrais avoir les deux premières lignes de votre fichier de config (ce sont les lignes # yaml-language-server et # debug, merci!

**Reply by anisite** [reply]
Je crois que vous devez encore avoir notre fichier de schéma que nous avions peut être envoyé au début, essayez de remplacer l'URL du schéma dans la ligne commentaire au début du fichier

 Exemple:
````yaml
# yaml-language-server: $schema=https://formulaires.it.mtess.gouv.qc.ca/api/v1/SIS/ObtenirSchema/form
# debug: langue=auto, showAll=false
````

**Reply by j0ber** [reply]
"watch" n'est pas nécessaire dans votre cas. En fait vous ne mettez que ce que vous utilisez. Comme dans l'exemple. Il faudrait voir les 2 premières lignes de votre fichier de config comme l'a mentionné @anisite. Il faut pointer sur le bon schéma.

**Reply by GigMatt** [reply]
Bonjour,  je vous remercie pour ces détails et effectivement, nous n'avions pas les bonnes lignes en haut de nos formulaires. Le tout fut ajusté et votre script fonctionne.
